pkg/gameserverallocations/controller.go

```diff
@@ -133,11 +133,17 @@ func (c *Controller) processAllocationRequest(ctx context.Context, w http.Respon
 	if err != nil {
 		return err
 	}
-	if status, ok := result.(*metav1.Status); ok {
-		w.WriteHeader(int(status.Code))
+	var code int
+	switch obj := result.(type) {
+	case *metav1.Status:
+		code = int(obj.Code)
+	case *allocationv1.GameServerAllocation:
+		code = http.StatusCreated
+	default:
+		code = 200
```

Comment:
It looks like the old behavior was to not set a code in the header at all unless the result was a metav1.Status whereas now we always set a code in the header. Is it better to only set the code when necessary, and not send any code at all for a successful response?

Reply:
There are probably a few things here, which are fun ways in which golang handles headers, that I found along the way.

1. This likely solves a bug that nobody ran into. Which is, when we previously set the `WriteHeader` in the current version, since we're writing the client at this point, it kicks the encoding type to `text/plain` - even though, no matter what we want `application/json` for all things (and we set it later in the code logic, but it will be ignored). This will cause all the k8s clients and `kubectl` to just fail. I found this out the hard way when implementing this fix.
2. If you don't set a HTTP status code header - then by default, Go will make it 200 on first write to the client. So you end up in the same place as you would do normally, but in this case it's now explicit rather than implicit.

Although a good point is - I shouldn't have it as "200" - I should use `http.StatusOK`. I'll make that change now.

Reply:
Thanks for the explanation.